### PR TITLE
Add HAProxy instructions for TiDB Dashboard

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <hi@breeswish.org>

### What is changed, added or deleted? (Required)

Add instructions to deploy TiDB Dashboard behind HAProxy, as well as reorg some contents behind a collapse panel.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: dashboard/dashboard-ops-reverse-proxy.md [hunk 1]
```diff
@@ -27,63 +27,110 @@ http://192.168.0.123:2379/dashboard/
 
 > **注意：**
 >
-> 该功能仅在较新版本的 `tiup cluster` 部署工具中提供。可通过以下命令升级 `tiup cluster`：
+> 该功能仅在较新版本的 TiUP Cluster 部署工具中提供。
```
Comment (TomShawn) [inline dashboard/dashboard-ops-reverse-proxy.md:30]: 较新版本是什么版本呢？请明确指出。

File: dashboard/dashboard-ops-deploy.md [hunk 2]
```diff
@@ -47,12 +47,17 @@ http://192.168.0.123:2379/dashboard/
 
 > **注意：**
 >
-> 该功能仅在较新版本的 `tiup cluster` 部署工具中提供。可通过以下命令升级 `tiup cluster`：
+> 该功能仅在较新版本的 TiUP Cluster 部署工具中提供。
```
Comment (TomShawn) [inline dashboard/dashboard-ops-deploy.md:50]: 较新版本是什么版本呢？请明确指出。

File: dashboard/dashboard-ops-reverse-proxy.md [hunk 3]
```diff
@@ -27,63 +27,110 @@ http://192.168.0.123:2379/dashboard/
...
+以 TiUP 部署且希望运行在 `http://example.com:8033/foo/` 为例，可指定以下配置：
 
 ```yaml
 server_configs:
   pd:
     dashboard.public-path-prefix: /foo
 ```
 
+<details>
+  <summary>使用 TiUP 全新部署集群的具体修改配置步骤</summary>
```
Comment (TomShawn) [inline dashboard/dashboard-ops-reverse-proxy.md:122]:
```suggestion
  <summary>使用 TiUP 部署全新集群时修改配置</summary>
```

File: dashboard/dashboard-ops-reverse-proxy.md [hunk 4]
```diff
@@ -27,63 +27,110 @@ http://192.168.0.123:2379/dashboard/
...
+
 若要全新部署集群，可在 TiUP 拓扑文件 `topology.yaml` 中加入上述配置项后进行部署，具体步骤参阅 [TiUP 部署文档](/production-deployment-using-tiup.md#第-3-步编辑初始化配置文件)。
 
-对于已部署的集群：
+</details>
+
+<details>
+  <summary>使用 TiUP 对已部署的集群的具体修改配置步骤</summary>
```
Comment (TomShawn) [inline dashboard/dashboard-ops-reverse-proxy.md:129]:
```suggestion
  <summary>使用 TiUP 修改已部署集群的配置</summary>
```

**Comment by breeswish** [comment]
@TomShawn PTAL again , thanks!

**Pull request merged**